Deprecate case matching on null in 4.4 (#1555)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

----
point.withinBBox(point({x:1, y:1}), point({x:0, y:0}), point({x:2, y:2}))
----

a|
label:functionality[]
label:deprecated[]
[source, cypher, role="noheader"]
----
MATCH (n)
RETURN
CASE n.prop
    WHEN null THEN 'one'
    ELSE 'two'
END
----
Currently, if n.prop is null, 'one' would be returned.
a|

> The matching of null == null in CASE expressions is deprecated and in future versions will no longer be able to be matched on.

Reviewer: - == actually doesn't exist in Cypher, so it should be one =, I believe
- It might be worth to point out that a Cypher comparision would return false here and we want to be consistent.
```suggestion
Since null = anything returns false in Cypher, a when exression will no longer match in future versions.
```

> a|
> label:functionality[]
> label:deprecated[]
> [source, cypher, role="noheader"]
> ----
> MATCH (n)
> RETURN
> CASE n.prop
>     WHEN null THEN 'one'
>     ELSE 'two'
> END
> ----
> Currently, if n.prop is null, 'one' would be returned.
> a|
> Since null = null returns false in Cypher, a when expression will no longer match in future versions.
> Please use `IS NULL` instead.
> [source, cypher, role="noheader"]
> ----
> MATCH (n)
> RETURN
> CASE
>     WHEN n.prop IS NULL THEN 'one'
>     ELSE 'two'
> END
> ----

Reviewer 2: ```suggestion
a|
label:functionality[]
label:deprecated[]
[source, cypher, role="noheader"]
----
MATCH (n)
RETURN
CASE n.prop
    WHEN null THEN 'one'
    ELSE 'two'
END
----
a|
Currently, if `n.prop` is `null`, `'one'` would be returned.
Since `null = null` returns `false` in Cypher, a `WHEN` expression will no longer match in future versions.

Please use `IS NULL` instead:
[source, cypher, role="noheader"]
----
MATCH (n)
RETURN
CASE
    WHEN n.prop IS NULL THEN 'one'
    ELSE 'two'
END
----
```
